cmd/deploy/start_test.go

```diff
@@ -40,106 +43,135 @@ func (suite *DeployStartTestSuite) SetupTest() {
+			kind:        "kubernetes",
+			file:        testAppYamlStrInvalid,
+			contentType: "yaml",
+			expected: de.StartPipelineResponse{
+				PipelineID: "12345",
+			},
+			assertion: func(result []byte, expected de.StartPipelineResponse) error {
+				suite.Equal(expectedValidationError, string(result))
+				return nil
+			},
+		},
+		{
+			name:        "start pipeline yaml success lambda",
+			kind:        "lambda",
+			file:        testLambdaYamlStr,
+			contentType: "yaml",
+			expected: de.StartPipelineResponse{
+				PipelineID: "12345",
+			},
+			assertion: func(result []byte, expected de.StartPipelineResponse) error {
+				// verify there is no validation errors being shown to the user
+				o := string(result)
+				suite.NotContains(o, "YAML is NOT valid. See the following errors:")
+
+				// verify that response formats as expected
+				var received = FormattableDeployStartResponse{}
+				err := yaml.Unmarshal(result, &received)
+				if err != nil {
+					return err
+				}
+				suite.Equal(expected.PipelineID, received.DeploymentId, "they should be equal")
+				return nil
+			},
+		},
 	}
-	var received = FormattableDeployStartResponse{}
-	yaml.Unmarshal(output, &received)
-	suite.Equal(expected.PipelineID, received.DeploymentId, "they should be equal")
-}
 
-func (suite *DeployStartTestSuite) TestDeployStartYAMLFailValidation() {
-	expected := &de.StartPipelineResponse{
-		PipelineID: "12345",
-	}
-	err := registerResponder(expected, http.StatusAccepted)
-	if err != nil {
-		suite.T().Fatalf("TestDeployStartYAMLSuccess failed with: %s", err)
-	}
-	tempFile := util.TempAppFile("", "app", testAppYamlStrInvalid)
-	if tempFile == nil {
-		suite.T().Fatal("TestDeployStartYAMLSuccess failed with: Could not create temp app file.")
-	}
-	suite.T().Cleanup(func() { os.Remove(tempFile.Name()) })
-	outWriter := bytes.NewBufferString("")
-	cmd := getDeployCmdWithFileName(outWriter, tempFile.Name(), "yaml")
-	err = cmd.Execute()
-	if err != nil {
-		suite.T().Fatalf("TestDeployStartYAMLSuccess failed with: %s", err)
-	}
-	output, err := io.ReadAll(outWriter)
-	if err != nil {
-		suite.T().Fatalf("TestDeployStartYAMLSuccess failed with: %s", err)
+	for _, c := range cases {
+		suite.T().Run(c.name, func(t *testing.T) {
+			err := registerResponder(c.expected, http.StatusAccepted, c.kind)
+			if err != nil {
+				suite.T().Fatalf("%s failed with: %s", c.name, err)
+			}
+			tempFile := util.TempAppFile("", "app", c.file)
+			if tempFile == nil {
+				suite.T().Fatalf("%s failed with: Could not create temp app file.", c.name)
+			}
+			suite.T().Cleanup(func() {
+				err := os.Remove(tempFile.Name())
+				if err != nil {
+					suite.T().Fatalf("%s failed with: %s", c.name, err)
+				}
+			})
+			outWriter := bytes.NewBufferString("")
+			cmd := getDeployCmdWithFileName(outWriter, tempFile.Name(), c.contentType)
+			err = cmd.Execute()
+			if err != nil {
+				suite.T().Fatalf("TestDeployStartJsonSuccess failed with: %s", err)
```

Comment:
`TestDeployStartJsonSuccess` isn't accurate anymore, but it was never needed - the test logs will associate errors with the right test case.